core/kernel/dt.c

```diff
@@ -58,6 +58,15 @@ const struct dt_driver *__dt_driver_end(void)
 	return &__rodata_dtdrv_end;
 }
 
+bool dt_have_prop(const void *fdt, int offs, const char *propname)
+{
+	const void *prop;
+
+	prop = fdt_getprop(fdt, offs, propname, NULL);
+
+	return prop ? true : false;
```

Comment:
There's no need for the ternary operator here. Since `dt_have_prop()` returns `bool` the compiler is required to normalize the value to return either `true` or `false`.

Reply:
I am not sure, but let compiler to normalize a value to true or false maybe not good, different compilers may have different behaviors? gcc/llvm or else. I checked linux and freebsd code, they not use compiler to normalize the value. freebsd not use bool, see https://github.com/freebsd/freebsd/blob/master/sys/dev/ofw/openfirm.c#L371
Linux kernel 
```
static inline bool of_property_read_bool(const struct device_node *np,
                                           const char *propname)
  {
          struct property *prop = of_find_property(np, propname, NULL);
          return prop ? true : false;
  }
```

So, could we keep `prop ? true : false` here?

Reply:
`? true : false` is clearly useless here because the `bool` type in `optee_os/core` is always defined as `_Bool`, which is C99 standard.
There are reaons why the links you provide have the normalization to true/false (or 0/1):
- The FreeBSD function returns `int`, which is not `_Bool`,
- The Linux function is part of [`libfdt`](https://github.com/dgibson/dtc/tree/master/libfdt), which makes no assumption on the actual type behind `bool`.

Reply:
From the C99 standard:
*6.3.1.2 Boolean type
When any scalar value is converted to _Bool, the result is 0 if the value compares equal to 0; otherwise, the result is 1.*

Reply:
Thanks @jforissier for the explaination. I'll fix it in next version.